Week 3 Update

faransikandar opened the pull request:
COMPLETED:
- [ ] Developed normalized values for train/test
- [ ] Developed trend metric
- [ ] Cleaned and combined dataset for 10 years of train/test
- [ ] Trained/tested models on one-year for 1995 vs. 2005 and did simplified 2-epoch NN on 1995-2004 vs. 2005-2014
- [ ] Slightly improved inference output (it outputs specific products, rather than just their corresponding vectors)
- [ ] Began modularizing code slightly

TO DO:
- [ ] Combine all predictions into one easy to read table
- [ ] Develop accuracy/validation method
- [ ] Modularize code and break into separate files
- [ ] Try different resampling methodologies to fix class imbalance problem
- [ ] Define custom loss metric which down-weights 0 values
- [ ] Define trend-weighted pct of exports metric - use this as target
- [ ] Test how/if a third vector can be introduced into the model definition
- [ ] Train on country-country pairs

faransikandar committed:
Week 3 Update

- Developed normalized values for train/test
- Developed trend metric
- Cleaned and combined dataset for 10 years of train/test
- Trained/tested models on one-year for 1995 vs. 2005 and did simplified 2-epoch NN on 1995-2004 vs. 2005-2014
- Slightly improved inference output (it outputs specific products, rather than just their corresponding vectors)
- Began modularizing code slightly

zhampel reviewed (commented):
Hey @faransikandar, I think it's best to remove any `pip show package` type lines from a python script, instead relegating them to a something like `print(package.__version__)`.

faransikandar committed:
Code Modularization - Beginning

- Feature engineering in data_clean
- Improved modularization
- Iterated on models
